site/server/app/com/fortysevendeg/exercises/services/free/userprogressops.scala

```diff
@@ -33,6 +33,23 @@ class UserProgressOps[F[_]](implicit I: Inject[UserProgressOp, F], EO: ExerciseO
   def saveUserProgress(userProgress: SaveUserProgress.Request): Free[F, UserProgress] =
     Free.inject[UserProgressOp, F](UpdateUserProgress(userProgress))
 
+  def fetchMaybeUserProgress(user: Option[User]): Free[F, OverallUserProgress] = {
+    user.fold(anonymousUserProgress)(theUser ⇒ fetchUserProgress(theUser))
+  }
+
+  private[this] def anonymousUserProgress: Free[F, OverallUserProgress] =
+    EO.getLibraries.map({ libs ⇒
+      val libraries = libs.map(l ⇒ {
+        OverallUserProgressItem(
+          l.name,
```

Comment:
Specially here since we have values such as 0 or false which provide no context as to what they refer